app/forms/idv/api_image_upload_form.rb

```diff
@@ -0,0 +1,76 @@
+module Idv
+  class ApiImageUploadForm
+    include ActiveModel::Model
+    include ActionView::Helpers::TranslationHelper
+
+    IMAGE_KEYS = %i[
+      front
+      back
+      selfie
+    ].freeze
+
+    validates_presence_of :front
+    validates_presence_of :back
+    validates_presence_of :selfie, if: :liveness_checking_enabled?
+
+    validate :validate_images
+
+    def initialize(params, liveness_checking_enabled:)
+      @params = params
+      @liveness_checking_enabled = liveness_checking_enabled
+    end
+
+    # Normally we'd return FormResponse, but that has errors as a hash,
+    # where the proofer reponses have errors as an array. This is easier to compare
+    # with the proofer responses
+    # @return [DocAuthClient::Response]
+    def submit
+      DocAuthClient::Response.new(
+        success: valid?,
+        errors: errors.full_messages,
+      )
+    end
+
+    def liveness_checking_enabled?
+      @liveness_checking_enabled
+    end
+
+    IMAGE_KEYS.each do |image_key|
```

Comment:
Could these be defined explicitly? I don't feel strongly, but sometimes I struggle to find methods if I search for `def selfie` and can't find it 🙂

Reply:
Ya I was on the fence about this, since 3 is small enough to write out but also easy enough to not. I'll expand this

Reply:
Fixed in c8093512607d4c4fa51e06fa2ee83b62c1b18e60